src/components/NavigationBar/index.tsx

```diff
@@ -62,26 +71,124 @@ export const navLinks = (
   }
 ];
 
-const NavigationBar = ({
-  mobile,
-  signout,
-  toggle,
-  userName
-}: NavigationProps) => {
+const systemLinks = (
+  flags: Flags,
+  userGroups: string[],
+  isUserSet: boolean
+) => [
+  {
+    link: '/systems?table-type=all-systems',
+    label: 'allCMSSystems',
+    isEnabled: true
+  },
+  {
+    link: '/systems?table-type=my-systems',
+    label: 'mySystems',
+    isEnabled: true
+  },
+  {
+    link: '/systems?table-type=bookmarked-systems',
+    label: 'bookmarkedSystems',
+    isEnabled: true
+  },
+  {
+    link: '/system/request-type',
+    label: 'addNewSystem',
+    isEnabled: true
+  }
+];
+
+// Handler for updating state for any NavDropDownButton
+const onToggle = (
+  index: number,
+  setIsOpen: React.Dispatch<React.SetStateAction<boolean[]>>
+): void => {
+  setIsOpen(prevIsOpen => {
+    const newIsOpen = [false];
+    newIsOpen[index] = !prevIsOpen[index];
+    return newIsOpen;
+  });
+};
+
+const NavigationBar = ({ signout, userName }: NavigationProps) => {
   const { t } = useTranslation();
+
   const flags = useFlags();
+
   const { groups, isUserSet } = useSelector((state: AppState) => state.auth);
 
+  const mobileNavRef = useRef<HTMLDivElement | null>(null);
+  const systemNavRef = useRef<HTMLDivElement | null>(null);
+
+  const isMobile = useCheckResponsiveScreen('tablet');
+
+  const location = useLocation();
+
+  // Controls mobile side nav
+  const [expanded, setExpanded] = useState(false);
+
+  // Controls NavDropDownButton state for (currently one) any nav dropdown items
+  const [isOpen, setIsOpen] = useState([false]);
+
+  // Click handler to toggle mobile sidenav
+  const onClick = (): void => setExpanded(prvExpanded => !prvExpanded);
+  // Click handler to toggle systems nav closed
+  const onSystemOutsideClick = (): void => setIsOpen([false]);
+
+  // Detects click outside mobile navigation
+  useOutsideClick(mobileNavRef, onClick);
+  // Detects clicks outside systems dropdown nav
+  useOutsideClick(systemNavRef, onSystemOutsideClick);
```

Comment:
i wonder if the callback should not take any args? i noticed it always runs with `false` in the hook itself

Reply:
The hook needs to take both args.  The ref will define the component that needs to trigger the callback if a mouseclick occurs outside its range.  Maybe we can sync to clarify, I might be missing something!

Reply:
Whoops sorry, i meant the the second arg in `useOutsideClick` is being called  as a callback with `false` as the argument. 

```ts
if (ref?.current && !ref?.current?.contains(evt.target as HTMLElement)) {
        callback(false);
}
```

i wonder if it should just be `callback()` and leave it up to the caller what they want to do, because `callback` only ever runs on an outside click

Reply:
a bit out of scope for this PR though, i just noticed that when checking things out

Reply:
Oh haha, idk, I didn't even look inside the hook, that's been there forever.  Yeah probably doesn't need it